src/main/java/org/wso2/siddhi/plugins/idea/SiddhiParserDefinition.java

```diff
@@ -0,0 +1,551 @@
+import static org.wso2.siddhi.plugins.idea.grammar.SiddhiQLParser.SCOL;
+import static org.wso2.siddhi.plugins.idea.grammar.SiddhiQLParser.SECONDS;
+import static org.wso2.siddhi.plugins.idea.grammar.SiddhiQLParser.SELECT;
+import static org.wso2.siddhi.plugins.idea.grammar.SiddhiQLParser.SINGLE_LINE_COMMENT;
+import static org.wso2.siddhi.plugins.idea.grammar.SiddhiQLParser.SNAPSHOT;
+import static org.wso2.siddhi.plugins.idea.grammar.SiddhiQLParser.SPACES;
+import static org.wso2.siddhi.plugins.idea.grammar.SiddhiQLParser.STREAM;
+import static org.wso2.siddhi.plugins.idea.grammar.SiddhiQLParser.STRING;
+import static org.wso2.siddhi.plugins.idea.grammar.SiddhiQLParser.STRING_LITERAL;
+import static org.wso2.siddhi.plugins.idea.grammar.SiddhiQLParser.TABLE;
+import static org.wso2.siddhi.plugins.idea.grammar.SiddhiQLParser.TRIGGER;
+import static org.wso2.siddhi.plugins.idea.grammar.SiddhiQLParser.TRUE;
+import static org.wso2.siddhi.plugins.idea.grammar.SiddhiQLParser.UNEXPECTED_CHAR;
+import static org.wso2.siddhi.plugins.idea.grammar.SiddhiQLParser.UNIDIRECTIONAL;
+import static org.wso2.siddhi.plugins.idea.grammar.SiddhiQLParser.UPDATE;
+import static org.wso2.siddhi.plugins.idea.grammar.SiddhiQLParser.WEEKS;
+import static org.wso2.siddhi.plugins.idea.grammar.SiddhiQLParser.WINDOW;
+import static org.wso2.siddhi.plugins.idea.grammar.SiddhiQLParser.WITH;
+import static org.wso2.siddhi.plugins.idea.grammar.SiddhiQLParser.WITHIN;
+import static org.wso2.siddhi.plugins.idea.grammar.SiddhiQLParser.YEARS;
+
+/**
+ * Defines the implementation of the parser for the Siddhi language.
+ */
+public class SiddhiParserDefinition implements ParserDefinition {
+
+    private static final IFileElementType FILE = new IFileElementType(SiddhiLanguage.INSTANCE);
+
+    static {
+        PSIElementTypeFactory.defineLanguageIElementTypes(SiddhiLanguage.INSTANCE, SiddhiQLParser.tokenNames,
+                SiddhiQLParser.ruleNames);
+    }
+
+    public static final TokenSet KEYWORDS = PSIElementTypeFactory.createTokenSet(SiddhiLanguage.INSTANCE, STREAM,
+            DEFINE, FUNCTION, TRIGGER, TABLE, APP, FROM, PARTITION, WINDOW, SELECT, GROUP, BY, HAVING, INSERT,
+            DELETE, UPDATE, RETURN, EVENTS, INTO, OUTPUT, EXPIRED, CURRENT, SNAPSHOT, FOR, RAW, OF, AS, AT, OR, AND,
+            ON, IN, IS, NOT, WITHIN, WITH, BEGIN, END, NULL, EVERY, LAST, ALL, FIRST, JOIN, INNER, OUTER, RIGHT, LEFT,
+            FULL, UNIDIRECTIONAL, YEARS, MONTHS, WEEKS, DAYS, HOURS, MINUTES, SECONDS, MILLISECONDS, FALSE, TRUE,
+            STRING, INT, LONG, FLOAT, DOUBLE, BOOL, OBJECT, AGGREGATION, AGGREGATE, PER);
+
+    public static final TokenSet COMMENTS = PSIElementTypeFactory.createTokenSet(SiddhiLanguage.INSTANCE,
+            SINGLE_LINE_COMMENT, MULTILINE_COMMENT);
+    public static final TokenSet WHITESPACE = PSIElementTypeFactory.createTokenSet(SiddhiLanguage.INSTANCE,
+            SPACES);
```

Comment:
can be moved to previous lines. Fix issues similar to this in other places.

Reply:
Fixed in commit 3aa87a6